engine/table/src/main/java/io/deephaven/engine/table/impl/select/FilterSelectColumn.java

```diff
@@ -0,0 +1,271 @@
+    }
+
+    @Override
+    public final WritableColumnSource<?> newFlatDestInstance(final long size) {
+        return InMemoryColumnSource.getImmutableMemoryColumnSource(size, Boolean.class, null);
+    }
+
+    @Override
+    public boolean isRetain() {
+        return false;
+    }
+
+    @Override
+    public boolean isStateless() {
+        return filter.permitParallelization();
+    }
+
+    @Override
+    public FilterSelectColumn copy() {
+        return new FilterSelectColumn(destName, filter.copy());
+    }
+
+    @Override
+    public void validateSafeForRefresh(BaseTable<?> sourceTable) {
+        filter.validateSafeForRefresh(sourceTable);
+    }
+
+    private class FilterFormula extends Formula {
+        public FilterFormula() {
+            super(null);
+        }
+
+        @Override
+        public Boolean getBoolean(long rowKey) {
+            WritableRowSet filteredIndex = filter.filter(RowSetFactory.fromKeys(rowKey), rowSet, tableToFilter, false);
+            return filteredIndex.isNonempty();
+        }
+
+        @Override
+        public Boolean getPrevBoolean(long rowKey) {
+            WritableRowSet filteredIndex = filter.filter(RowSetFactory.fromKeys(rowKey), rowSet, tableToFilter, true);
+            return filteredIndex.isNonempty();
+        }
+
+        @Override
+        public Object get(final long rowKey) {
+            return getBoolean(rowKey);
+        }
+
+        @Override
+        public Object getPrev(final long rowKey) {
+            return getPrevBoolean(rowKey);
+        }
+
+        @Override
+        public ChunkType getChunkType() {
+            return ChunkType.Object;
+        }
+
+        @Override
+        public FillContext makeFillContext(final int chunkCapacity) {
+            return FILL_CONTEXT_INSTANCE;
+        }
+
+        @Override
+        public void fillChunk(
+                @NotNull final FillContext fillContext,
+                @NotNull final WritableChunk<? super Values> destination,
+                @NotNull final RowSequence rowSequence) {
+            doFill(rowSequence, destination, false);
+        }
+
+        @Override
+        public void fillPrevChunk(
+                @NotNull final FillContext fillContext,
+                @NotNull final WritableChunk<? super Values> destination,
+                @NotNull final RowSequence rowSequence) {
+            doFill(rowSequence, destination, true);
+        }
+
+        private void doFill(@NotNull RowSequence rowSequence, WritableChunk<? super Values> destination,
+                boolean usePrev) {
+            final WritableObjectChunk<Boolean, ?> booleanDestination = destination.asWritableObjectChunk();
+            booleanDestination.setSize(rowSequence.intSize());
+            try (final RowSet inputRowSet = rowSequence.asRowSet();
+                    final RowSet filtered = filter.filter(inputRowSet, rowSet, tableToFilter, usePrev);
```

Comment:
You need to use the previous full set...
```suggestion
                    final RowSet filtered = filter.filter(inputRowSet, usePrev ? rowSet.prev() : rowSet, tableToFilter, usePrev);
```